crates/polars-core/src/frame/group_by/aggregations/dispatch.rs

```diff
@@ -109,19 +111,29 @@ impl Series {
         use DataType::*;
 
         match self.dtype() {
+            Boolean => self.cast(&Float64).unwrap().agg_median(groups),
             Float32 => SeriesWrap(self.f32().unwrap().clone()).agg_median(groups),
             Float64 => SeriesWrap(self.f64().unwrap().clone()).agg_median(groups),
-            dt if dt.is_numeric() || dt.is_temporal() => {
-                let ca = self.to_physical_repr();
-                let physical_type = ca.dtype();
-                let s = apply_method_physical_integer!(ca, agg_median, groups);
-                if dt.is_logical() {
-                    // back to physical and then
-                    // back to logical type
-                    s.cast(physical_type).unwrap().cast(dt).unwrap()
-                } else {
-                    s
-                }
+            dt if dt.is_numeric() => {
+                apply_method_physical_integer!(self, agg_median, groups)
+            },
+            #[cfg(feature = "dtype-date")]
+            Date => {
+                let s = self.cast(&Int64).unwrap() * (US_IN_DAY as f64);
```

Comment:
Why the multiplication here? We can just use the physical type right?

And shouldn't this return `Date`?

Reply:
on the return type, there was a bit of discussion here https://github.com/pola-rs/polars/pull/12670#pullrequestreview-1749268415

I think it can make sense to upcast `Date` to `Datetime('us')`, as they're limited to the same range of dates (+/- 262,000 years since unix epoch)

Reply:
@ritchie46 it's in the OP, the same way the mean/median of ints is a float, the mean/median of dates is now a datetime.